odoo/orm/fields_relational.py

```diff
@@ -100,8 +104,8 @@ def validated(domain):
     _description_relation = property(attrgetter('comodel_name'))
     _description_context = property(attrgetter('context'))
 
-    def _description_domain(self, env):
-        domain = self.domain(env[self.model_name]) if callable(self.domain) else self.domain  # pylint: disable=not-callable
+    def _description_domain(self, env) -> str | list:
```

Comment:
Should it be easier to always return an str for the webclient and doesn't have this double API. It means that we will be able to override `_description_domain` directly for One2many toward Many2oneReference :thinking:

Reply:
The goal here is just to document it, 18.1 is now stable so I don't want to change this.